core/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointService.java

```diff
@@ -32,33 +33,41 @@
  */
 public class LocalCheckpointService extends AbstractIndexShardComponent {
 
+    public static String MAX_SEQ_NO = "max_seq_no";
+    public static String LOCAL_CHECKPOINT_KEY = "local_checkpoint";
+
     /**
      * we keep a bit for each seq No that is still pending. to optimize allocation, we do so in multiple arrays
      * allocating them on demand and cleaning up while completed. This setting controls the size of the arrays
      */
     public static Setting<Integer> SETTINGS_BIT_ARRAYS_SIZE = Setting.intSetting("index.seq_no.checkpoint.bit_arrays_size", 1024,
         4, Setting.Property.IndexScope);
 
-
     /**
      * an ordered list of bit arrays representing pending seq nos. The list is "anchored" in {@link #firstProcessedSeqNo}
      * which marks the seqNo the fist bit in the first array corresponds to.
      */
     final LinkedList<FixedBitSet> processedSeqNo;
     final int bitArraysSize;
-    long firstProcessedSeqNo = 0;
+    long firstProcessedSeqNo;
 
     /** the current local checkpoint, i.e., all seqNo lower (&lt;=) than this number have been completed */
-    volatile long checkpoint = SequenceNumbersService.NO_OPS_PERFORMED;
+    volatile long checkpoint;
 
     /** the next available seqNo - used for seqNo generation */
-    volatile long nextSeqNo = 0;
+    volatile long nextSeqNo;
 
+    public LocalCheckpointService(final ShardId shardId, final IndexSettings indexSettings) {
```

Comment:
I pushed a5b99c65259b933c4b379e29e160685ee427eda9 and 6e8f11ae3da78c4b82cd267c1daba3951b462eee.